app/models/CategoryRecord.scala

```diff
@@ -0,0 +1,101 @@
+import pages.{AssessmentPage, HasSupplementaryUnitPage, SupplementaryUnitPage}
+import play.api.libs.json.{Json, OFormat}
+import queries.RecordCategorisationsQuery
+
+final case class CategoryRecord(
+  eori: String,
+  recordId: String,
+  category: Int,
+  supplementaryUnit: Option[Int] = None,
+  measurementUnit: Option[String] = None
+)
+
+object CategoryRecord {
+
+  implicit lazy val format: OFormat[CategoryRecord] = Json.format
+
+  def build(answers: UserAnswers, eori: String, recordId: String): EitherNec[ValidationError, CategoryRecord] =
+    (
+      getCategory(answers, recordId),
+      answers.getOptionalPageValue(answers, HasSupplementaryUnitPage, SupplementaryUnitPage),
+      getMeasurementUnit(answers)
+    ).parMapN((category, supplementaryUnit, measurementUnit) =>
+      CategoryRecord(
+        eori = eori,
+        recordId = recordId,
+        category = category,
+        supplementaryUnit = supplementaryUnit,
+        measurementUnit = measurementUnit
+      )
+    )
+
+  def getCategory(answers: UserAnswers, recordId: String): EitherNec[ValidationError, Int] =
+    answers.get(RecordCategorisationsQuery) match {
+      case Some(recordCategorisations) =>
+        recordCategorisations.records.get(recordId) match {
+          case Some(categorisationInfo) => chooseCategory(recordId, answers, categorisationInfo)
+          case _                        => Left(NonEmptyChain.one(RecordIdMissing(RecordCategorisationsQuery)))
+        }
+      case _                           => Left(NonEmptyChain.one(PageMissing(RecordCategorisationsQuery)))
+    }
+
+  def chooseCategory2Or3(
+    recordId: String,
+    answers: UserAnswers,
+    categorisationInfo: CategorisationInfo
+  ): EitherNec[ValidationError, Int] = {
+    val category2AssessmentsCount = categorisationInfo.categoryAssessments.count(_.category == 2)
+    if (category2AssessmentsCount > 0) {
+      answers.getPageValue(AssessmentPage(recordId, categorisationInfo.categoryAssessments.length - 1)) match {
+        case Right(AssessmentAnswer.NoExemption) => Right(2)
+        case Right(_)                            => Right(3)
+        case _                                   => Right(2)
+      }
+    } else {
+      Right(3)
+    }
+  }
+
+  def chooseCategory(
+    recordId: String,
+    answers: UserAnswers,
+    categorisationInfo: CategorisationInfo
+  ): EitherNec[ValidationError, Int] = {
+    val category1AssessmentsCount = categorisationInfo.categoryAssessments.count(_.category == 1)
+    if (category1AssessmentsCount > 0) {
+      answers.getPageValue(
+        AssessmentPage(recordId, category1AssessmentsCount - 1)
+      ) match {
+        case Right(AssessmentAnswer.NoExemption) => Right(1)
+        case Right(_)                            => chooseCategory2Or3(recordId, answers, categorisationInfo)
+        case _                                   => Right(1)
+      }
+    } else {
+      chooseCategory2Or3(recordId, answers, categorisationInfo)
```

Comment:
can't have category 2 assessments without cat 1 I think? would this not then be a category 3?